pkg/dumper/dump.go

```diff
@@ -208,30 +211,72 @@ func (c PostgresDumpHandler) Dump(ctx context.Context, r io.Reader, path string)
 		line martaapi.Line
 		dir  martaapi.Direction
 	}
-	corrections := map[string]correction{}
+
+	var (
+		// As we loop over stuff, we're going to keep track of _all_
+		// the various line, direction, and station names we come across.
+		// Then, we'll loop over those and resolve the database IDs of
+		// their respective records by fuzzy-matching.
+		seenLineNames      = map[string]struct{}{}
+		seenDirectionNames = map[string]struct{}{}
+		seenStationNames   = map[string]struct{}{}
+
+		lineNameResolutions      = map[string]*uint{}
+		directionNameResolutions = map[string]*uint{}
+		stationNameResolutions   = map[string]*uint{}
```

Comment:
Since any name we need to look up is probably repeated dozens of times in the API body, we'll gather them in de-duped form (`seen*Names`), and then look up the third-rail database ID for each one (using fuzzy matches with aliases).